source/patterns/@aws-solutions-constructs/aws-kinesisstreams-gluejob/lib/index.ts

```diff
@@ -0,0 +1,186 @@
+   * 	"FieldSchema": [{
+   *  	"name": "id",
+   *  	"type": "int",
+   *    "comment": "Identifier for the record"
+   *  }, {
+   *    "name": "name",
+   *    "type": "string",
+   *    "comment": "The name of the record"
+   *  }, {
+   *    "name": "type",
+   *    "type": "string",
+   *    "comment": "The type of the record"
+   *  }, {
+   *    "name": "numericvalue",
+   *    "type": "int",
+   *    "comment": "Some value associated with the record"
+   *  },
+   */
+  readonly fieldSchema?: CfnTable.ColumnProperty [];
+  /**
+   * Glue Database for this construct. If not provided the construct will create a new Glue Database.
+   * The database is where the schema for the data in Kinesis Data Streams is stored
+   */
+  readonly database?: CfnDatabase;
+  /**
+   * Glue Table for this construct, If not provided the construct will create a new Table in the
+   * database. This table should define the schema for the records in the Kinesis Data Streams.
+   * Either @table or @fieldSchema is mandatory. If @table is provided then @fieldSchema is ignored
+   */
+  readonly table?: CfnTable;
+  /**
+   * The output data stores where the transformed data should be written. Current supported data stores
+   * include only S3, other potential stores may be added in the future.
+   */
+  readonly outputDataStore?: defaults.SinkDataStoreProps;
+}
+
+export class KinesisStreamGlueJob extends Construct {
+  public readonly kinesisStream: Stream;
+
+  public readonly glueJob: [ CfnJob, IRole ];
+
+  public readonly database: CfnDatabase;
+
+  public readonly table: CfnTable;
+
+  public readonly outputBucket!: [Bucket, (Bucket | undefined)?];
+
+  constructor(scope: Construct, id: string, props: KinesisStreamGlueJobProps) {
+    super(scope, id);
+
+    this.kinesisStream = defaults.buildKinesisStream(this, {
+      existingStreamObj: props.existingStreamObj,
+      kinesisStreamProps: props.kinesisStreamProps,
+    });
+
+    if (props.database !== undefined) {
+      this.database = props.database!;
+    } else {
+      this.database = defaults.createGlueDatabase(scope);
+    }
+
+    if (props.fieldSchema === undefined && props.table === undefined) {
+      throw Error('Either fieldSchema or table property has to be set, both cannot be optional');
+    }
+
+    if (props.table !== undefined) {
+      this.table = props.table;
+    } else {
+      this.table = defaults.createGlueTable(scope, this.database, props.fieldSchema!, 'kinesis', {
+        STREAM_NAME: this.kinesisStream.streamName
+      });
+    }
+
+    this.glueJob = defaults.buildGlueJob(this, {
+      existingCfnJob: props.existingGlueJob,
+      glueJobProps: props.glueJobProps,
+      table: this.table!,
+      database: this.database!,
+      outputDataStore: props.outputDataStore!
+    });
+
+    this.buildRolePolicy(scope, this.database, this.table, this.glueJob[1]);
```

Comment:
`buildRolePolicy(...): IRole` returns `IRole` and it's not assigned to anything, was it missed?

Reply:
Agreed, updated code